### PR TITLE
protocol change for exporting TLF keys

### PR DESCRIPTION
**Pull request by songgao**
Part 1 of [CORE-3524](https://keybase.atlassian.net/browse/CORE-3524); wanted to get this merged first so other changes could include vendored `protocol` package.

### CODE REVIEW

File: protocol/avdl/tlf_keys.avdl [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    // crypt keys of all generations for this TLF, in order, starting from
+    // FirstValidKeyGen
+    array<Bytes32> CryptKeys;
+  }
+
+  /**
+   getTLFCryptKeys returns TLF crypt keys from all generations.
+   */
+  TLFCryptKeys getTLFCryptKeys(string tlfName);
```
Comment (maxtaco) [inline protocol/avdl/tlf_keys.avdl]: Why two different rpcs that do largely the same thing? Also why two files?

Comment (songgao) [inline protocol/avdl/tlf_keys.avdl]: One is from service to kbfs (TlfKeys) and the other from cli to service (Tlf).

I was thinking the one from cli to service (Tlf) might include other Tlf related stuff in the future, but not sure if there's ever gonna be. Should I just use the same interface?

**Comment by maxtaco** [comment]
Got it, ok!

👍

**Comment by maxtaco** [comment]
Test failure is a known flake

File: protocol/avdl/tlf.avdl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+@namespace("keybase.1")
+
```
Comment (maxtaco) [inline protocol/avdl/tlf.avdl]: a comment describing when to use this protocol?

File: protocol/avdl/tlf_keys.avdl [hunk 3]
```diff
@@ -0,0 +1,29 @@
+@namespace("keybase.1")
+
+protocol tlfKeys {
```
Comment (maxtaco) [inline protocol/avdl/tlf_keys.avdl]: same, provide a comment to say how different from the other protocol?

**Comment by maxtaco** [comment]
add a few more comments but otherwise 👍

**Comment by songgao** [comment]
Cool; Thanks Max!

**Pull request merged**